libbeat/processors/conditions/equals.go

```diff
@@ -0,0 +1,109 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package conditions
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type equalsValue struct {
+	Int  uint64
+	Str  string
+	Bool bool
+}
+
+// EqualsCondition is a Condition for testing string equality.
+type EqualsCondition map[string]equalsValue
```

Comment:
Would also call it `Equals` here and leave out the `condition` part as it's part of the package name. If someone instantiates it from outside it reads `conditions.NewEquals`

Reply:
++